src/main/drivers/bus_i2c_stm32f30x.c

```diff
@@ -231,15 +231,17 @@ bool i2cWrite(uint8_t addr_, uint8_t reg, uint8_t data)
         }
     }
 
-    /* Send Register address */
-    I2C_SendData(I2Cx, (uint8_t) reg);
+    if (reg != -1) {
```

Comment:
>The code works because it calls next I2C_TransferHandling unconditionally (without sending configured byte).
Flags are different (TXIS is sent until NBYTES is transferred, TCR is not sent), but ignored.

Yeah, I just noticed the same while giving a second look on flags. Will fix and test it (knowing the overhaul is around the corner).